client/state/data-layer/wpcom/sites/scan/enqueue/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Internal dependencies
+ */
+import { registerHandlers } from 'state/data-layer/handler-registry';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	JETPACK_SCAN_ENQUEUE_UPDATE,
+	JETPACK_SCAN_ENQUEUE_REQUEST,
+	JETPACK_SCAN_ENQUEUE_REQUEST_SUCCESS,
+	JETPACK_SCAN_ENQUEUE_REQUEST_FAILURE,
+} from 'state/action-types';
+
+// NOTE: This is currently a mock endpoint.
+// It always returns the following payload:
+// ```
+// { "enqueue": true }
+// ```
+const enqueueScan = ( action ) => {
+	return http(
+		{
+			apiNamespace: 'wpcom/v2',
+			method: 'POST',
+			path: `/sites/${ action.siteId }/scan/enqueue`,
+			body: {},
+		},
+		action
+	);
+};
+
+const onEnqueueScanSuccess = ( action, enqueueResult ) => {
+	return [
+		{
+			type: JETPACK_SCAN_ENQUEUE_REQUEST_SUCCESS,
+			siteId: action.siteId,
+		},
+		{
+			type: JETPACK_SCAN_ENQUEUE_UPDATE,
+			siteId: action.siteId,
+			payload: enqueueResult,
+		},
```

Comment:
This doesn't really matter at this point, but I think we can move the logic that request the Scan state to this place. So, if enqueueing a Scan run succeeds, we dispatch the request for the Scan state. This way, we only dispatch the request if we know the first request has already finished. Here is an example of that: https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom/sites/scan/index.js#L84

Reply:
I definitely thought about moving the scan status request here, but I hesitated because I wasn't sure if maybe we had a use case for starting a scan but not getting the status afterward.

On a more technical note, would I still need a delay for the status request (or be able to write one) if I added the action to the list here?

Reply:
On a real setup, no delay is required because if you move it there, it will only run after the first request finishes.

Let keep it as you did and change it later when we know how things are going to work.